arch/rv32i/src/epmp.rs

```diff
@@ -612,9 +612,7 @@ impl<const MAX_AVAILABLE_REGIONS_OVER_TWO: usize> kernel::platform::mpu::MPU
 
         let region = PMPRegion::new_app(start as *const u8, size, permissions);
 
-        if region.is_none() {
-            return None;
-        }
+        region?;
```

Comment:
```suggestion
        let _ = region?;
```
Maybe that could be a more natural way to write this? I agree with @ppannuto that an expression consisting of a variable name followed by a `?`, while silently discarding the unpacked value, is not a particularly great way to implement an early-abort. However, it is quite natural to use a `?` for this purpose to extract the contained value at the end of more generic, complex expressions. Maybe `let _ =` can be used to make it clear that we're not actually interested in the contained value here.

Reply:
Aren't you rewriting this anyway?

Reply:
I'm actually more okay with this 'line of code on its own with a question mark'. It's clear that the `?` has to be doing _something_ otherwise why would this line even be here?

It's the stuff like this:

    let region = PMPRegion::new_app(start as *const u8, size, permissions)?;

where that little, easy-to-miss `?` hanging out at the end magically injects very easily overlooked control flow that I think is a an absolute crime and a terrible anti-feature.

Reply:
> Aren't you rewriting this anyway?

Well, yes. I was just trying to see whether @ppannuto's concerns could've been addressed by a `let _ =` binding.

Reply:
I guess, yeah, the "Rust-y" version would be

```
let region = PMPRegion::new_app(start as *const u8, size, permissions)?;
```